### PR TITLE
New Presence: AMC+

### PR DESCRIPTION
**Pull request by thepoliticallycorrect**
New Presence for the AMC streaming service. https://www.amcplus.com/

Handles 3 states: browsing catalogue, watching series and watching a movie.

Browsing catalogue:
image

Watching Series:
image

Watching a movie:
image

Watching a movie (paused):
image

Might add buttons in a posterior version.

### CODE REVIEW

File: websites/A/AMCPlus/presence.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      ),
+      live = timestamps[1] === Infinity;
+
+    smallImageText = live
+      ? (await strings).live : video.paused
+        ? (await strings).pause : (await strings).play;
+
+    smallImageKey = live ? "live" : video.paused ? "pause" : "play";
+
+    startTimestamp = live ? elapsed : timestamps[0];
```
Comment (Slowlife01) [inline websites/A/AMCPlus/presence.ts:79]:
```suggestion
    if (live) startTimestamp = elapsed;
```

File: websites/A/AMCPlus/presence.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ * @param {Number} videoDuration Video duration seconds
+ */
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (Slowlife01) [inline websites/A/AMCPlus/presence.ts:23]: Use Presence.getTimestamps() instead

File: websites/A/AMCPlus/presence.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  if (smallImageText !== undefined) 
+    data.smallImageText = smallImageText;
+  
+  if (startTimestamp !== undefined) 
+    data.startTimestamp = startTimestamp;
+  
+  if (endTimestamp !== undefined) 
+    data.endTimestamp = endTimestamp;
+  
+  presence.setActivity(data, video ? !video.paused : true);
```
Comment (Slowlife01) [inline websites/A/AMCPlus/presence.ts:108]:
```suggestion
  presence.setActivity(data);
```

File: websites/A/AMCPlus/presence.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      state = slot3.textContent;
+    } else {
+      // A movie only has slot1 (the title)
+      state = "Watching movie";
+    }
+
+    const timestamps = presence.getTimestamps(
+        Math.floor(video.currentTime),
+        Math.floor(video.duration)
+      ),
```
Comment (Bas950) [inline websites/A/AMCPlus/presence.ts:56]: just use getMediaTimestamps

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**